plugins/registry.ts

```diff
@@ -62,64 +78,85 @@ const standardizeRoute = (route) => {
         return dispatchPluginComponentAction('LeftSidebarHeader', this.id, component);
-    }
+    });
 
     // Register a component fixed to the bottom of the team sidebar. Does not render if
     // user is only on one team and the team sidebar is not shown.
     // Accepts a React component. Returns a unique identifier.
-    registerBottomTeamSidebarComponent(component) {
+    registerBottomTeamSidebarComponent = reArg(['component'], ({component}: DPluginComponentProp) => {
         return dispatchPluginComponentAction('BottomTeamSidebar', this.id, component);
-    }
+    });
 
     // Register a component fixed to the bottom of the post message.
     // Accepts a React component. Returns a unique identifier.
-    registerPostMessageAttachmentComponent(component) {
+    registerPostMessageAttachmentComponent = reArg(['component'], ({component}: DPluginComponentProp) => {
         return dispatchPluginComponentAction('PostMessageAttachment', this.id, component);
-    }
+    });
 
     // Register a component to show as a tooltip when a user hovers on a link in a post.
     // Accepts a React component. Returns a unique identifier.
     // The component will be passed the following props:
     // - href - The URL for this link
     // - show - A boolean used to signal that the user is currently hovering over this link. Use this value to initialize your component when this boolean is true for the first time, using `componentDidUpdate` or `useEffect`.
-    registerLinkTooltipComponent(component) {
+    registerLinkTooltipComponent = reArg(['component'], ({component}: DPluginComponentProp) => {
         return dispatchPluginComponentAction('LinkTooltip', this.id, component);
-    }
+    });
 
     // Add a button to the channel header. If there are more than one buttons registered by any
     // plugin, a dropdown menu is created to contain all the plugin buttons.
     // Accepts the following:
     // - icon - React element to use as the button's icon
     // - action - a function called when the button is clicked, passed the channel and channel member as arguments
-    // - dropdown_text - string or React element shown for the dropdown button description
-    // - tooltip_text - string or React element shown for tooltip appear on hover
-    registerChannelHeaderButtonAction(icon, action, dropdownText, tooltipText) {
+    // - dropdownText - string or React element shown for the dropdown button description
+    // - tooltipText - string or React element shown for tooltip appear on hover
+    registerChannelHeaderButtonAction = reArg([
+        'icon',
+        'action',
+        'dropdownText',
+        'tooltipText',
+    ], ({
+        icon,
+        action,
+        dropdownText,
+        tooltipText,
+    }: {
+        icon: ReactResolvable;
+        action: PluginComponent['action'];
+        dropdownText: ReactResolvable;
+        tooltipText: ReactResolvable;
+    }) => {
```

Comment:
Amazing work on this file! Since this is now TS, I wonder if we can somehow export `PluginRegistry` through `packages/types` for plugins to import?

Reply:
Long term goal, for sure! This is a step towards that end, but still some work to do before that, I believe.